backend/selectgen.ml

```diff
@@ -773,23 +803,51 @@ method insert_op_debug env op dbg rs rd =
 method insert_op env op rs rd =
   self#insert_op_debug env op Debuginfo.none rs rd
 
-(* Add the instructions for the given expression
-   at the end of the self sequence *)
+method insert_endregions env regions =
+  match regions with
+  | [] -> ()
+  | regions ->
+     (* Coalesce multiple simultaneous Iendregion *)
+     let final_region = List.hd (List.rev regions) in
+     self#insert env (Iop Iendregion) final_region [| |]
+
+method insert_endregions_until env ~suffix regions =
+  if suffix != regions then
```

Comment:
Maybe push the equivalent of this check into `strip_suffix` and then just let the `[]` case deal with this above?